Source/WebCore/Modules/permissions/PermissionStatus.cpp

```diff
@@ -54,6 +55,12 @@ static HashMap<MainThreadPermissionObserverIdentifier, std::unique_ptr<MainThrea
     return map;
 }
 
+static void createAssociatedMainThreadPermissionObserver(WeakPtr<PermissionStatus, WeakPtrImplWithEventTargetData>&& permissionStatus, ScriptExecutionContextIdentifier contextIdentifier, PermissionState state, PermissionDescriptor descriptor, PermissionQuerySource source, WeakPtr<Page>&& page, ClientOrigin&& origin, MainThreadPermissionObserverIdentifier identifier) 
```

Comment:
We may not need a separate function here since my suggestion below avoid code duplication.